runtime/src/main/java/org/corfudb/comm/ChannelImplementation.java

```diff
@@ -0,0 +1,72 @@
+package org.corfudb.comm;
+
+import io.netty.channel.Channel;
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.ServerChannel;
+import io.netty.channel.epoll.Epoll;
+import io.netty.channel.epoll.EpollEventLoopGroup;
+import io.netty.channel.epoll.EpollServerSocketChannel;
+import io.netty.channel.epoll.EpollSocketChannel;
+import io.netty.channel.kqueue.KQueue;
+import io.netty.channel.kqueue.KQueueEventLoopGroup;
+import io.netty.channel.kqueue.KQueueServerSocketChannel;
+import io.netty.channel.kqueue.KQueueSocketChannel;
+import io.netty.channel.local.LocalChannel;
+import io.netty.channel.local.LocalServerChannel;
+import io.netty.channel.nio.NioEventLoopGroup;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.channel.socket.nio.NioSocketChannel;
+import java.util.concurrent.ThreadFactory;
+import javax.annotation.Nonnull;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+/** An enum representing channel implementation types available to the client. */
+@AllArgsConstructor
+public enum ChannelImplementation {
+    /** Automatically select best channel type (EPOLL/KQUEUE if available, otherwise
+     *  fallback to NIO).
+     */
+    AUTO(Epoll.isAvailable() ? EpollSocketChannel.class :
+        KQueue.isAvailable() ? KQueueSocketChannel.class :
+            NioSocketChannel.class,
+        Epoll.isAvailable() ? EpollServerSocketChannel.class :
+            KQueue.isAvailable() ? KQueueServerSocketChannel.class :
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3358)